mistralrs-core/src/sampler.rs

```diff
@@ -12,9 +12,12 @@ use pyo3::pyclass;
 
 use rand::distributions::{Distribution, WeightedIndex};
 use rand_isaac::Isaac64Rng;
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
 use serde::{Deserialize, Serialize};
 use tokenizers::Tokenizer;
 
+const SEQUENCE_BREAKERS: &[&str] = &["\n", ":", "\\", "*"];
```

Comment:
I assume `Vec<String>` is only used elsewhere to avoid lifetime annotation with the struct? (_I haven't gone through the entire PR_)

This suggestion might not make much of a difference, but you could initialize this once here with `OnceLock` / `OnceCell` (Rust 1.70), or with Rust 1.80 there is also `LazyLock` now available:

```suggestion
use std::sync::LazyLock;
static SEQUENCE_BREAKERS: LazyLock<Vec<String>> = LazyLock::new(|| {
    ["\n", ":", "\\", "*"].map(String::from).to_vec()
});
```

You then have it initialized once on first access and can deref for the inner type to access the `Vec<String>` with `*SEQUENCE_BREAKERS`.

Then later if you do need to store a copy, you can use `unwrap_or(*SEQUENCE_BREAKERS.clone())`, or just access the static variable since it's already `Vec<String>`.